### PR TITLE
Added support to adding simple block rules

### PR DESCRIPTION
**Pull request by KostyaTretyak**
```js
const marked = require('marked');
const blockStr = `
# Example usage with embed block code

@@@ gist
a9dfd77500990871fc58b97fdb57d91f.js
@@@

@@@ youtube
JgwnkM5WwWE
@@@
`;

marked.setBlockRule(/^@@@ *(\w+)\n([\s\S]+?)\n@@@/, function (execArr) {
  const channel = execArr[1];
  switch (channel) {
    case 'youtube': {
        const id = marked.escape(execArr[2]);
        return `<iframe width="420" height="315" src="https://www.youtube.com/embed/${id}"></iframe>\n`;
     }
    case 'gist': {
        const id = marked.escape(execArr[2]);
        return `<script src="https://gist.github.com/${id}"></script>\n`;
     }
  }
});
const html = marked.parse(blockStr);
console.log(html);
```

Output:

```html
<h1 id="example-usage-with-embed-block-code">Example usage with embed block code</h1>
<script src="https://gist.github.com/a9dfd77500990871fc58b97fdb57d91f.js"></script>
<iframe width="420" height="315" src="https://www.youtube.com/embed/JgwnkM5WwWE"></iframe>
```

### CODE REVIEW

**Comment by joshbruce** [comment]
Thanks @KostyaTretyak! Gonna flag this for milestone 0.5.0. If you could take a look at #1002 - I believe it could cause this to have merge conflicts in the future as we would need to add tests to document and validate the solution works. Cheers.

**Comment by KostyaTretyak** [comment]
@joshbruce  no, it does not seem that my changes were in conflict with the changes introduced in #1002.

File: lib/marked.js [hunk 1]
```diff
@@ -408,6 +414,18 @@ Lexer.prototype.token = function(src, top, bq) {
       continue;
     }
 
+    // simple rules
+    if (Lexer.simpleRules.length) {
+      var simpleRules = Lexer.simpleRules;
+      for (var i = 0; i < simpleRules.length; i++) {
+        if (cap = simpleRules[i].exec(src)) {
+          src = src.substring(cap[0].length);
+          this.tokens.push({ type: 'simpleRule' + simpleRules.length, cap: cap });
```
[the remark this answers is not in the document]
Comment (KostyaTretyak) [reply lib/marked.js]: Yes, you are right, I'll correct it right now. Thank you.

**Comment by joshbruce** [comment]
See #1026 

Closing as merge conflicts - also building out the issue above to consolidate ideas and approaches proposed.

**Pull request closed without merging**